### PR TITLE
Brotli support for HTTP interfaces

### PR DESCRIPTION
**Pull request by fandyushin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
Brotli support for HTTP interfaces:
`Content-Encoding: br`

### CODE REVIEW

File: cmake/find_brotli.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    find_library (BROTLI_LIBRARY libbrotli)
+    find_path (BROTLI_INCLUDE_DIR NAMES decode.h encode.h port.h types.h PATHS ${BROTLI_INCLUDE_PATHS})
+endif ()
+
+if (BROTLI_LIBRARY AND BROTLI_INCLUDE_DIR)
+elseif (NOT MISSING_INTERNAL_BROTLI_LIBRARY)
+    set (BROTLI_INCLUDE_DIR ${ClickHouse_SOURCE_DIR}/contrib/libbrotli/c/include)
+    set (USE_INTERNAL_BROTLI_LIBRARY 1)
+    set (BROTLI_LIBRARY libbrotli)
+endif ()
```
Comment (proller) [inline cmake/find_brotli.cmake]: set(USE_BROTLI 1)

**Comment by alexey-milovidov** [comment]
Looks good.
Need to add a test.

**Comment by fandyushin** [comment]
I have several problems:
For run test i need brotli util, it's present in ubuntu repo for example. 
Clang wont build brotli sources, gcc - ok
I cant beat cmake - `target_include_directories(clickhouse_common_io SYSTEM BEFORE PRIVATE ${BROTLI_INCLUDE_DIR})`  doesent work, but `target_include_directories(clickhouse_common_io PUBLIC ${BROTLI_INCLUDE_DIR})` - ok

**Comment by fandyushin** [comment]
its seems that I'm done

File: .gitmodules [hunk 2]
```diff
@@ -63,4 +63,10 @@
 	url = https://github.com/ClickHouse-Extras/libgsasl.git
 [submodule "contrib/cppkafka"]
 	path = contrib/cppkafka
-	url = https://github.com/ClickHouse-Extras/cppkafka.git
+	url = https://github.com/mfontanini/cppkafka.git
```
Comment (alexey-milovidov) [inline .gitmodules]: Unrelated changes.

**Comment by alexey-milovidov** [comment]
Ok. Now it is implemented for data insertion.

How hard is to implement `BrotliWriteBuffer` and to use it when `enable_http_compression` is enabled and client said that it accepts `br`? And probably to provide `http_brotli_compression_level` setting.

Do you want to implement it in next pull request?

**Comment by fandyushin** [comment]
I've fixed .gitmodules. It changes too often :)
I think its not complicated to implement write buffer. I can do it.
